site/source/docs/api_reference/emscripten.h.rst

```diff
@@ -482,8 +482,9 @@ Functions
 
 .. c:function:: void emscripten_exit_with_live_runtime(void)
 
-  Exits the program immediately, but leaves the runtime alive so that you can continue to run code later (so global destructors etc., are not run). Note that the runtime is kept alive automatically when you do an asynchronous operation like :c:func:`emscripten_async_call`, so you don't need to call this function for those cases.
+  Stops the current thread of execution, but leaves the runtime alive so that you can continue to run code later (so global destructors etc., are not run). Note that the runtime is kept alive automatically when you do an asynchronous operation like :c:func:`emscripten_async_call`, so you don't need to call this function for those cases.
 
+  In a multithreaded application, this just exits the current thread (and allows running code later in the Web Worker in which it runs).
```

Comment:
Ops, sorry, this has slipped my attention.

Originally the design was indeed to do just like what happens with the main thread: the main/pthread_main yields/unwinds out, but remains alive for later event-based calls into it.

Now reading the docs wording, I find that the original function name `emscripten_exit_with_live_runtime()` is maybe a bit misleading: saying that it both "exits" and also keeps a "live runtime" seem to contradict each other.

Here `In a multithreaded application, this just exits the current thread` the word `exit` might be understood by the reader to mean the same as in quit, stop or terminate. But of course the thread (in web, thread == the Worker) is kept alive, and it will execute any events that are passed to it.

Reply:
Yes we should probably replace "exits" here with something closer to "unwinds"

Reply:
We might want to draw a distinction between "thread" here and "worker".    In the current models workers basically never exit and never stop responding to message, but they can stop hosting a pthread and be returned the pool.     "emscripten_exit_with_live_runtime" means we continue to host a pthread on the worker.

I'm not sure if other types of worker (i.e. audio worklets) have this concept since IIRC they can already never die to be recycled?   Perhaps `emscripten_exit_with_live_runtime()` should error if called from another type of worker or that is not hosting a pthread (including the main pthread that that set)

Reply:
The current implementation of `emscripten_exit_with_live_runtime()` is though compatible with Audio Worklets and Wasm Workers in that they can be used to achieve the unwind effect. So we might let users do that to be able to reuse/share code.

Reply:
I see.  So under Audio Worklets or Wasm Workers this is exactly the same as `emscripten_unwind_to_js_event_loop`? 

May be worth mentioning that here too.